src/runtime/naming/name.cpp

```diff
@@ -261,16 +261,29 @@ namespace hpx { namespace naming
         void id_type_impl::preprocess_gid(serialization::output_archive& ar) const
         {
             // unmanaged gids do not require any special handling
+            // check-pointing does not require any special handling here neither
             if (unmanaged == type_)
             {
                 return;
             }
 
+            // we should not call this function during check-pointing operations
+            if (ar.try_get_extra_data<checkpointing_tag>() != nullptr)
+            {
+                // this is a check-pointing operation, we do not support this
+                HPX_THROW_EXCEPTION(invalid_status,
+                    "id_type_impl::preprocess_gid",
+                    "can't check-point managed id_type's, use a component "
+                    "client instead");
+            }
+
             auto& split_gids = ar.get_extra_data<
                 serialization::detail::preprocess_gid_types>();
 
             if (split_gids.has_gid(*this))
             {
+                // archive doesn't support credit splitting or no splitting
+                // required
```

Comment:
This comment is not correct. The gid has been split already and we don't need to do anything further.

Reply:
Good catch, thanks; it's fixed on master now.